pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -175,66 +182,193 @@ pub mod pallet {
+
+			let current_asset_id = NextAssetId::<T>::get();
+
+			let (currency_ids, next_asset_id) = Self::generate_sequential_asset_ids(current_asset_id, currency_length)?;
+
+			let pool_id = T::PoolId::from(currency_ids.blake2_256());
+
+			// Todo: change that.
+			T::DepositCurrency::hold(
+				&T::RuntimeHoldReason::from(HoldReason::Deposit),
+				&who,
+				T::BaseDeposit::get().saturating_add(
+					T::DepositPerCurrency::get()
+						.saturating_mul(currency_length.saturated_into())
+						.saturated_into(),
+				),
+			)?;
+
+			currencies
+				.iter()
+				.enumerate()
+				.try_for_each(|(idx, entry)| -> DispatchResult {
+					let asset_id: &T::AssetId = currency_ids.get(idx).ok_or(Error::<T>::IndexOutOfBounds)?;
+					T::Fungibles::create(asset_id.clone(), pool_id.clone().into(), false, entry.min_balance)?;
+
+					// set metadata for new asset class
+					T::Fungibles::set(
+						asset_id.clone(),
+						&pool_id.clone().into(),
+						entry.name.clone().into(),
+						entry.symbol.clone().into(),
+						denomination,
+					)?;
+
+					Ok(())
+				})?;
+
+			// Touch the pool account in order to be able to transfer the collateral currency to it
+			T::CollateralCurrency::touch(T::CollateralAssetId::get(), &pool_id.clone().into(), &who)?;
+
+			Pools::<T>::set(
+				&pool_id,
+				Some(PoolDetails::new(who, checked_curve, currency_ids, transferable)),
+			);
+
+			// update the storage for the next tx.
+			NextAssetId::<T>::set(next_asset_id);
+
+			Self::deposit_event(Event::PoolCreated { id: pool_id });
+
+			Ok(())
+		}
+
+		#[pallet::call_index(2)]
+		#[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().writes(1))]
+		pub fn reset_team(
+			origin: OriginFor<T>,
+			pool_id: T::PoolId,
+			team: PoolManagingTeam<AccountIdOf<T>>,
+			currency_idx: u32,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			let pool_details = Pools::<T>::get(&pool_id).ok_or(Error::<T>::UnknownPool)?;
+
+			ensure!(pool_details.is_manager(&who), Error::<T>::NoPermission);
+
+			let asset_id = pool_details
+				.bonded_currencies
+				.get(currency_idx as usize)
+				.ok_or(Error::<T>::IndexOutOfBounds)?;
+
+			let pool_id_account = pool_id.into();
+
+			T::Fungibles::reset_team(
+				asset_id.to_owned(),
+				pool_id_account.clone(),
+				team.admin,
+				pool_id_account,
+				team.freezer,
+			)
 		}
 
-		#[pallet::call_index(1)]
+		#[pallet::call_index(3)]
+		#[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().writes(1))]
+		pub fn reset_manager(
+			origin: OriginFor<T>,
+			pool_id: T::PoolId,
+			manager: Option<AccountIdOf<T>>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			Pools::<T>::try_mutate(pool_id, |maybe_entry| -> DispatchResult {
+				if let Some(entry) = maybe_entry {
+					ensure!(entry.is_manager(&who), Error::<T>::NoPermission);
+					entry.manager = manager;
+					Ok(())
+				} else {
+					Err(Error::<T>::UnknownPool.into())
+				}
```

Comment:
works but isn't this more readable:

```suggestion
				let entry = maybe_entry.as_mut().ok_or(Error::<T>::UnknownPool)?;
				ensure!(entry.is_manager(&who), Error::<T>::NoPermission);
				entry.manager = manager;
				Ok(()) // This may actually be optional
```

Reply:
love it: [30cbeb3](https://github.com/KILTprotocol/kilt-node/pull/768/commits/30cbeb34d5d6e3ae8a7561c38beef14206755a5d)